src/main/java/bio/terra/service/tabulardata/google/BigQueryProject.java

```diff
@@ -148,7 +148,24 @@ public boolean deleteTable(String datasetName, String tableName) {
 
   public void updateDatasetAcls(Dataset dataset, List<Acl> acls) throws InterruptedException {
     DatasetInfo datasetInfo = dataset.toBuilder().setAcl(acls).build();
-    AclUtils.aclUpdateRetry(() -> bigQuery.update(datasetInfo));
+    AclUtils.aclUpdateRetry(
+        () -> {
+          try {
+            bigQuery.update(datasetInfo);
+          } catch (BigQueryException ex) {
+            bigQueryAclUpdateShouldRetry(ex);
+          }
+          return null;
+        });
+  }
+
+  private void bigQueryAclUpdateShouldRetry(RuntimeException ex) {
+    String message = ex.getMessage();
+    if (message.startsWith("IAM setPolicy") && message.endsWith("does not exist.")) {
```

Comment:
Is there an error code we could check instead of just the error message?

Reply:
Hmm okay I was thinking they would return a 404 error code or something to distinguish that from permissions errors